Solid modifications

themrcesi opened the pull request:
Okay guys, after almost more than 15 hours working... @alvarezGarciaMarcos and I finally achieved everything related with SOLID.

Now, we are able to share routes modifying permissions, uploading comments, uploading images, view shared routes with all the linked content...you know, everything.
If you want to try it, first of all remove /viade in your POD, be sure of removing everything inside please.

@alvarogarinf 's way of reading images perfectly worked.

So now, tests are failing, only a few of them. 
- @InigoGutierrez can you fix the tests related to SOLID? The ones you created some time ago. Now they are behaving as expected, so it would be easy and fast to adapt them.
- We need to fix also the other tests.

@lamasumas can you check all the internationalization "errors" and fix them?

We need also to check css of the new files and so on... In the meating we talk about this.

themrcesi committed:
Structure & Routing

themrcesi committed:
Basic structure

themrcesi committed:
Nav bar

themrcesi committed:
WebId updated in profile

Commit:
Added Notifications styling

Commit:
Merge branch 'notifications' of https://github.com/Arquisoft/viade_en1b into notifications

Commit:
Finished with the styling

alvarogarinf committed:
Functionality to add images and videos to the route
Added the front-end

Commit:
Trying to fix solid

Commit:
Folders permissions working fine

Commit:
Step by step

Commit:
Another step closer

Commit:
Notification parser

Commit:
Merge branch 'notifications' into solid_modifications

Commit:
Added notifications read from solid

Commit:
Notifications working

alvarogarinf committed:
Merge branch 'dev' of https://github.com/Arquisoft/viade_en1b into dev_alvaro

alvarogarinf committed:
Added the functionality to upload images and videos in each button,
but it is not uploaded to solid, so it cannot show the images in MyRoutes component

Commit:
Sharing routes is now possible

Sharing routes between users is now possible, also, when a user wants to
delete a route shared by another user, the application allows to delete
the shared route from the users pod.

Commit:
Permission for comments are set

Commit:
The author is the username with the provider

themrcesi committed:
Authority correctly checked

Commit:
Comments working

Commit:
Modified comments

Commit:
Modified permissions on writting comments

themrcesi committed:
Comment author

Commit:
All functionality is working (almost) perfectly

Commit:
Media should be working

Commit:
Everything works fine, only tests are left

themrcesi committed:
Warnings removed

themrcesi committed:
UnShare test redux

themrcesi committed:
Minor change

themrcesi committed:
Fixing tests (more)

Marcos guapo

lamasumas committed:
Error in local fixed

themrcesi committed:
More fixes

alvarogarinf committed:
Now, it is mandatory for the user to fill at least the name and upload a route to get to push the button

InigoGutierrez committed:
Tests fixed. Some more added. Cleaned comments and logs.

InigoGutierrez committed:
Removed unused variable.

InigoGutierrez committed:
When uploading a route the comments should be an array.

InigoGutierrez committed:
Comments in route to upload back to a string and not a vector.

InigoGutierrez committed:
Merge pull request #146 from Arquisoft/solid_modifications_testing

Tests fixed. Some more added. Cleaned comments and logs.

lamasumas committed:
Merge remote-tracking branch 'origin/solid_modifications' into edu_stuff

lamasumas committed:
Notifications translated

alvarogarinf committed:
Merge branch 'solid_modifications' of https://github.com/Arquisoft/viade_en1b into solid_modifications_alvaro

alvarogarinf committed:
Now, you cannot upload a route file that is not a .gpx and if you try an alert will be launched and
you could not push the submit button
I have also fixed a bug that occurs when you try to upload a file and then push cancel

themrcesi committed:
Merge branch 'solid_modifications_testing' into solid_modifications

themrcesi committed:
Tests added

alvarogarinf committed:
Merge pull request #145 from Arquisoft/solid_modifications_alvaro

Required fields to upload a route: name & file

InigoGutierrez committed:
Added JSDoc to routes.js.

themrcesi committed:
RouteDetails tests

alvarogarinf committed:
Added tests for comments

alvarogarinf committed:
Merge branch 'solid_modifications' of https://github.com/Arquisoft/viade_en1b into solid_modifications_alvaro

themrcesi committed:
Warnings fixed

InigoGutierrez committed:
Removed a console.log call.

themrcesi committed:
Stupid parser tests

lamasumas committed:
Merge pull request #144 from Arquisoft/edu_stuff

Added missing words to locale

alvarogarinf committed:
Merge pull request #149 from Arquisoft/solid_modifications_alvaro

Added tests for comments

themrcesi committed:
Merge pull request #150 from Arquisoft/solid_modifications_cesar

More test

themrcesi committed:
Just two console.log removed

themrcesi committed:
Merge branch 'solid_modifications' into solid_modifications_cesar

lamasumas committed:
Runtime view updated

themrcesi committed:
HotFix: Number of own/shared routes fixed

themrcesi committed:
Tests updated

themrcesi committed:
Minor improvements

themrcesi committed:
Merge pull request #152 from Arquisoft/solid_modifications_cesar

HotFix: Number of own/shared routes fixed

themrcesi committed:
Redux diagram updated

themrcesi committed:
Building block diagrams updated

themrcesi committed:
Building block view section updated

themrcesi committed:
Quality scenarios and Technical Risks

themrcesi committed:
Merge branch 'solid_modifications' into modals_pelayo

themrcesi committed:
Removed unused files

themrcesi committed:
Domain concepts updated

themrcesi committed:
Technical debt: videos

lamasumas committed:
Merge pull request #151 from Arquisoft/edu_stuff

Runtime view updated

InigoGutierrez committed:
Merge pull request #148 from Arquisoft/solid_modifications_routes_JSdoc

Added JSDoc to routes.js.

InigoGutierrez committed:
Documented arrows as flow of information and changed a couple of them on Business Context diagram.

InigoGutierrez committed:
Added disadvantages of Codacy to Decisions part.

lamasumas committed:
Technical debt row added

InigoGutierrez committed:
Added checks for wrong files in routes folder.

InigoGutierrez committed:
Added checks for wrong shared routes file.

InigoGutierrez committed:
Now also check processing of the urls of routes in the shared routes file.

InigoGutierrez committed:
Fixed a couple mistakes.

alvarezGarciaMarcos committed:
Fixed uploadButton and ShareRoute tests

Commit:
Modifying styling of notifications

alvarezGarciaMarcos committed:
Merge pull request #155 from Arquisoft/fix-tests-marcos

Fixed uploadButton and ShareRoute tests

themrcesi committed:
HOTFIX

themrcesi committed:
Fixed warning

Commit:
Merged

themrcesi committed:
Fix

themrcesi committed:
Merge branch 'solid_modifications' into handling_unconventional_pods

themrcesi committed:
Another try

Commit:
Added styling for notifications

Commit:
Added styling for Comments and slideshow

InigoGutierrez committed:
Added return null to getRouteFromPod so it can be checked after it is called.

InigoGutierrez committed:
Enhanced drawbacks of TDD in Decisions section.

InigoGutierrez committed:
Explained routes module testing.

InigoGutierrez committed:
Merge pull request #154 from Arquisoft/handling_unconventional_pods

Handling unconventional pods

InigoGutierrez committed:
A bunch of changes suggested by Codacy.

InigoGutierrez committed:
Reverted hooks.js file; it seems its != shouldn't be changed to !==.

InigoGutierrez reviewed (approved):
So many green ticks! 😍
I think we can merge the remaining PRs into this one and finally merge into dev so we can make the final changes from there.

themrcesi committed:
Merge pull request #153 from Arquisoft/solid_modifications_cesar

Documentation

themrcesi committed:
Path fixed

themrcesi committed:
Fixed console error

lamasumas committed:
Some codacy upgrades

lamasumas committed:
Fixed an error

lamasumas committed:
Fixed some test error

Commit:
Add styling for the Settings page

Commit:
More styling and fixing

I added our favicon to the page, and also fixed some errors in the
Settings tests regarding some changes done to the settings page

lamasumas committed:
Some codacy improvements

lamasumas committed:
Some other codes improvements

alvarogarinf committed:
JsDoc in:
- src/parser/parser.js
- src/utils/functions.js

themrcesi committed:
Route is cleared when view is changed.

themrcesi committed:
Also in dropdowns

alvarogarinf committed:
Added comments to the following components
	src/components/routes/myRoutes/MyRoutes.js
	src/components/routes/routeDetails/RouteDetails.js
	src/components/routes/routeList/RouteList.js
	src/components/routes/routeSummary/RouteSummary.js
	src/components/routes/shareRoute/ShareRoute.js
	src/components/routes/uploadRoute/UploadRoute.js
	src/components/routes/uploadRoute/uploadButton/UploadButton.js

themrcesi committed:
Delete solid

Now when a route is deleted all its linked data: comments file and resources are also deleted.

themrcesi committed:
Merge branch 'solid_modifications' into documentation_inigo

themrcesi committed:
Selenium and Artillery

alvarogarinf committed:
Added some more little descriptions for some react components

alvarogarinf committed:
Merge pull request #159 from Arquisoft/solid_modifications_alvaro

JsDoc in:

Commit:
Added styling to routeDetails so everything is shown

Commit:
Merge branch 'fix-final-styling' of https://github.com/Arquisoft/viade_en1b into fix-final-styling

garciaAlvarezPelayo committed:
_allmodals_plus_tests

themrcesi committed:
Conflicts solved

garciaAlvarezPelayo committed:
Merge branch 'fix-final-styling' into modals_pelayo

garciaAlvarezPelayo committed:
allmodals_tests_internationalization

lamasumas committed:
Merge pull request #157 from Arquisoft/edu_stuff

Solid modifications codacy

garciaAlvarezPelayo committed:
modals_tests_internationalization

themrcesi committed:
H

Commit:
Fixed some styling related to the _accept all_ button

themrcesi committed:
Merge pull request #161 from Arquisoft/modals_pelayo

Modals pelayo

themrcesi committed:
Merge pull request #156 from Arquisoft/documentation_inigo

Documentation updates
